features/yield/FarmingCalculator.tsx

```diff
@@ -0,0 +1,172 @@
+  const { empState } = EmpState.useContainer();
+  const { usdPrice, pool } = Balancer.useContainer();
+  const [umaPrice, setUmaPrice] = useState<number | null>(4.0);
+  const [yUSDAdded, setyUSDAdded] = useState<string | null>("1000");
+  const [USDCAdded, setUSDCAdded] = useState<string | null>("1000");
+  const [fracLiquidity, setFracLiquidity] = useState<string | null>(null);
+  const [umaYieldAmount, setUmaYieldAmount] = useState<string | null>(null);
+  const [usdYieldAmount, setUsdYieldAmount] = useState<string | null>(null);
+  const [apr, setApr] = useState<string | null>(null);
+
+  getUmaPrice().then((price) => {
+    setUmaPrice(price);
+  });
+  //   console.log("PRICE", umaPrice);
+
+  const calculateYield = () => {
+    if (!yUSDAdded || Number(yUSDAdded) < 0) {
+      return null;
+    }
+    if (!USDCAdded || Number(USDCAdded) < 0) {
+      return null;
+    }
+    if (!usdPrice || Number(usdPrice) < 0) {
+      return null;
+    }
+    if (!pool) {
+      return null;
+    }
+    if (!umaPrice) {
+      return null;
+    }
+    const totalValueAddedToPool =
+      Number(USDCAdded) + Number(yUSDAdded) * Number(usdPrice);
+    if (!totalValueAddedToPool || totalValueAddedToPool == 0) {
+      return null;
+    }
+    const fracLiquidityProvided =
+      totalValueAddedToPool / Number(pool.liquidity);
+    setFracLiquidity((fracLiquidityProvided * 100).toFixed(4));
+    const weeklyUMARewarded = 25000;
+    const umaPaidPerWeek = fracLiquidityProvided * weeklyUMARewarded;
+    setUmaYieldAmount(umaPaidPerWeek.toFixed(4));
+    const usdYieldPerWeek = umaPaidPerWeek * umaPrice;
+    setUsdYieldAmount(usdYieldPerWeek.toFixed(4));
+
+    const apr =
+      ((umaPaidPerWeek * WEEKS_PER_YEAR * umaPrice) / totalValueAddedToPool) *
+      100;
+    setApr(apr.toFixed(4));
+  };
+
+  // Update the yield whenever the parameters change.
+  useEffect(() => {
+    calculateYield();
+  }, [usdPrice, yUSDAdded, USDCAdded]);
+
+  return (
+    <span>
+      <Typography variant="h5">$UMA Farming calculator</Typography>
+      <Typography>
+        During the liquidity mining program 25k $UMA rewards will be paid out to
+        LP providers in the yUSD balancer pool. Rewards are calculated as a
+        pro-rata contribution to the liquidity pool. To learn more about the
+        liquidity mining program see UMA Medium post{" "}
+        <a
+          href="https://medium.com/uma-project/liquidity-mining-on-uma-is-now-live-5f6cb0bd53ee"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          here
+        </a>
+        .
+      </Typography>
+
+      <Box pt={2}>
+        <form noValidate autoComplete="off">
+          <Grid container spacing={2}>
+            <Grid item md={4} sm={6} xs={12}>
+              <FormInput>
+                <TextField
+                  fullWidth
+                  type="number"
+                  label="yUSD added to pool"
+                  value={yUSDAdded?.toString() || ""}
```

Comment:
You should follow the pattern we now have in every other component where we refactor all the null checking to the front of the function and don't have to do it at all in the JSX components